stdpopsim/engines.py

```diff
@@ -112,6 +112,15 @@ def _warn_zigzag(self, demographic_model):
                 "https://github.com/popsim-consortium/stdpopsim/issues/745"
             )
 
+    def _warn_mutation_rate_mismatch(self, contig, demographic_model):
+        if demographic_model.mutation_rate is not None and demographic_model.mutation_rate != contig.mutation_rate:
+            warnings.warn(
+                "The demographic model mutation rate differs from the contig mutation "
+                "rate (simulation uses the contig mutation rate). Diversity levels "
+                "may be different than expected for this species. For details see: "
+                "(url of some documentation for this)"
```

Comment:
maybe stick in the mutation rates? Like `f"The demographic model has mutation rate {demographic_model.mutation_rate}, but this simulation used mutation rate {contig.mutation_rate}. ..."

Reply:
Good idea. Updated now to include those rates